routesrv/polling.go

```diff
@@ -142,10 +144,57 @@ func (p *poller) process(routes []*eskip.Route) []*eskip.Route {
 		routes = cloner.Do(routes)
 	}
 
+	disabledFilters := len(p.disabledFilters) != 0
+
+	routes = p.validateRoutes(routes, disabledFilters)
+
 	// sort the routes, otherwise it will lead to different etag values for the same route list for different orders
 	sort.SliceStable(routes, func(i, j int) bool {
 		return routes[i].Id < routes[j].Id
 	})
 
 	return routes
 }
+
+func (p *poller) validateRoutes(routes []*eskip.Route, disableFilters bool) []*eskip.Route {
+	validRoutes := make([]*eskip.Route, 0, len(routes))
+
+	var disabledFiltersMap map[string]bool
+
+	if disableFilters {
+		disabledFiltersMap = toBooleanMap(p.disabledFilters)
+	}
+
+	var validRouteFilters bool
+	for _, r := range routes {
+		validRouteFilters = true
+		for _, filter := range r.Filters {
+			if filter.Name == predicates.PathSubtreeName || filter.Name == predicates.PathName || filter.Name == predicates.HostName || filter.Name == predicates.PathRegexpName || filter.Name == predicates.MethodName || filter.Name == predicates.HeaderName || filter.Name == predicates.HeaderRegexpName {
+				validRouteFilters = false
+				log.Errorf("trying to use %q as filter, but it is only available as predicate", filter.Name)
+				break
+			}
+
+			if disableFilters {
+				if _, ok := disabledFiltersMap[filter.Name]; ok {
+					validRouteFilters = false
+					log.Errorf("trying to use %q filter, which is disabled", filter.Name)
+					break
+				}
+			}
+		}
+		if validRouteFilters {
+			validRoutes = append(validRoutes, r)
+		}
+	}
+
+	return validRoutes
+}
+
+func toBooleanMap[C comparable](values []C) map[C]bool {
```

Comment:
I would suggest to use struct{} instead of bool, which saves memory and you do not use the bool value anyways in line 179.
```
func toMap[C comparable](values []C) map[C]struct{} {
	h := make(map[C]struct{}, len(values))
	for _, value := range values {
		h[value] = struct{}{}
	}
	return h
}
```